airbyte-integrations/connectors/source-stripe/source_stripe/streams.py

```diff
@@ -286,12 +327,30 @@ def path(self, **kwargs):
     def request_params(self, stream_slice: Mapping[str, Any] = None, **kwargs):
         params = super().request_params(stream_slice=stream_slice, **kwargs)
         params["subscription"] = stream_slice["subscription_id"]
+
+        # add 'starting_after' param
+        if not params.get("starting_after") and stream_slice and stream_slice.get("starting_after"):
+            params["starting_after"] = stream_slice["starting_after"]
+
         return params
 
     def read_records(self, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs) -> Iterable[Mapping[str, Any]]:
         subscriptions_stream = Subscriptions(authenticator=self.authenticator, account_id=self.account_id, start_date=self.start_date)
-        for subscriptions in subscriptions_stream.read_records(sync_mode=SyncMode.full_refresh):
-            yield from super().read_records(stream_slice={"subscription_id": subscriptions["id"]}, **kwargs)
+        for subscription in subscriptions_stream.read_records(sync_mode=SyncMode.full_refresh):
+
+            items_obj = subscription.get("items", {})
+            if not items_obj:
+                continue
+
+            items = items_obj.get("data", [])
+
+            # get the next pages with subscription items
+            items_next_pages = []
+            if items_obj.get("has_more") and items:
+                stream_slice = {"subscription_id": subscription["id"], "starting_after": items[-1]["id"]}
+                items_next_pages = super().read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice, **kwargs)
+
+            yield from [*items, *items_next_pages]
```

Comment:
Same comment here, we make this `read` method universal for at least 3 streams, the only difference I can see here are: nested data objects and field names, which could be handled by input variables based on stream.